relacy/thread.hpp

```diff
@@ -233,7 +233,7 @@ struct thread_info : thread_info_base
                 for (thread_id_t i = 0; i != thread_count; ++i)
                 {
                     timestamp_t acq_rel_order2 = acq_rel_order_[i];
-                    if (acq_rel_order2 >= rec.last_seen_order_[i])
+                    if (acq_rel_order2 >= rec.first_seen_order_[i])

```

Comment:
So this is what fixes the test case. This seem to make sense from all perspectives: if we are checking something here then it should be first_seen_order_, checking last_seen_order_ just doesn't make sense.

Please add a comment here along the lines of (I know that I wasn't very generous with comments in the past, but at least some traces will be useful):
```
// This check ensures read-read coherence, 1.10/16:
//	If a value computation A of an atomic object M happens before a value
//	computation B of M, and A takes its value from a side effect X on M,
//	then the value computed by B shall either be the value stored by X or
//	the value stored by a side effect Y on M, where Y follows X in the
//	modification order of M.
```

Reply:
Will do!